Src/API/ActionBuilder.lua

```diff
@@ -0,0 +1,109 @@
+local Sift = require(script.Parent.Parent.Packages.Sift)
+
+--[[
+	Metrik SDK ActionBuilder & Action
+]]
+
+--[=[
+	@class ActionBuilder
+
+	ActionBuilder enables developers to build sophisticated/complicated Actions.
+
+	The ActionBuilder class follows the Builder design pattern; https://refactoring.guru/design-patterns/builder
+]=]
+local ActionBuilder = {}
+
+--[=[
+	@class Action
+
+	Action enable developers to interact with the Metrik backend, when an Action is instantiated the backend is notified so that servers
+		that have this Action is callable from the Metrik site.
+
+	You can only create an Action, through an ActionBuilder.
+]=]
+local Action = {}
+
+Action.Instantiated = {}
+Action.Prototype = {}
+
+ActionBuilder.Public = {}
+ActionBuilder.Prototype = {}
+
+function Action.Prototype:CanRun(...: unknown) -- todo: figure out how the 'middleware' impl is going to look.
+	return true
+end
+
+function Action.Prototype:OnRun(...: unknown)
+	print(`Method ':OnRun' has been called on Action<"{self.Name}">`)
+end
+
+function ActionBuilder.Prototype.SetName(self: ActionBuilder, actionName: string): ActionBuilder
```

Comment:
We should set a max length limit for the name, I forgot what it's set to in the API (or if it's even there.)

Reply:
there wasn't a limit :smile:

Reply:
set it at like 50 chars and forget it imo